crates/utils/src/settings/structs.rs

```diff
@@ -1,118 +1,168 @@
-use regex::Regex;
-use serde::Deserialize;
+use doku::Document;
+use serde::{Deserialize, Serialize};
 use std::net::{IpAddr, Ipv4Addr};
 
-#[derive(Debug, Deserialize, Clone, SmartDefault)]
+#[derive(Debug, Deserialize, Serialize, Clone, SmartDefault, Document)]
 #[serde(default)]
 pub struct Settings {
+  /// settings related to the postgresql database
   #[serde(default)]
   pub database: DatabaseConfig,
   #[default(Some(RateLimitConfig::default()))]
+  /// rate limits for various user actions, by user ip
   pub rate_limit: Option<RateLimitConfig>,
+  /// Settings related to activitypub federation
   #[default(FederationConfig::default())]
   pub federation: FederationConfig,
   #[default(CaptchaConfig::default())]
   pub captcha: CaptchaConfig,
+  /// Email sending configuration. All options except login/password are mandatory
   #[default(None)]
   pub email: Option<EmailConfig>,
+  /// Parameters for automatic configuration of new instance (only used at first start)
   #[default(None)]
   pub setup: Option<SetupConfig>,
+  /// the domain name of your instance (mandatory)
   #[default("unset")]
+  #[doku(example = "example.com")]
   pub hostname: String,
+  /// Address where lemmy should listen for incoming requests
   #[default(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)))]
+  #[doku(as = "String")]
   pub bind: IpAddr,
+  /// Port where lemmy should listen for incoming requests
   #[default(8536)]
   pub port: u16,
+  /// Whether the site is available over TLS. Needs to be true for federation to work.
   #[default(true)]
   pub tls_enabled: bool,
+  /// Address where pictrs is available (for image hosting)
   #[default(None)]
+  #[doku(example = "http://localhost:8080")]
   pub pictrs_url: Option<String>,
+  /// Regex for slurs which are prohibited. Example: `(\bThis\b)|(\bis\b)|(\bsample\b)`
   #[default(None)]
   pub additional_slurs: Option<String>,
+  /// Maximum length of local community and user names
   #[default(20)]
   pub actor_name_max_length: usize,
-  #[default(None)]
-  #[serde(skip)]
-  pub webfinger_community_regex: Option<Regex>,
-  #[default(None)]
-  #[serde(skip)]
-  pub webfinger_username_regex: Option<Regex>,
```

Comment:
doku macro failed on these fields, so i had to move them out of the config struct and into lazy_static instead.